src/fides/api/ops/service/connectors/consent_email_connector.py

```diff
@@ -97,20 +135,48 @@ def test_connection(self) -> Optional[ConnectionTestStatus]:
     def needs_email(
```

Comment:
`Saasconnector.needs_email` fires on the privacy request runner when we check if a privacy request needs to exit in a state of `awaiting_email_send` because we batch these emails and only send them once a week. I've modified whether we need to send a consent email to add in all the new checks, see: `filter_privacy_preferences_for_propagation`.  Note that the old workflow is still maintained here.